web_src/js/features/comp/WebHookEditor.js

```diff
@@ -1,11 +1,69 @@
 import $ from 'jquery';
 const {csrfToken} = window.config;
 
+const initAuthenticationHeaderSection = function() {
+  const $authHeaderSection = $('.auth-headers');
+
+  if ($authHeaderSection.length === 0) {
+    return;
+  }
+
+  const $checkbox = $authHeaderSection.find('.checkbox input');
+
+  const updateHeaderContentType = function() {
+    const isBasicAuth = $authHeaderSection.find('#auth_header_type').val() === 'basic';
+    const $basicAuthFields = $authHeaderSection.find('.basic-auth');
+    const $tokenAuthFields = $authHeaderSection.find('.token-auth');
+
+    if (isBasicAuth) {
+      $basicAuthFields.addClass('required');
+      $basicAuthFields.find('input').attr('required', '');
```

Comment:
I think `$basicAuthFields.find('input').prop('required', true)` would be the canonical way to set boolean attributes with jQuery.